reana_server/status.py

```diff
@@ -341,12 +361,285 @@ def get_status(self):
+            logging.error("Error while calling `metrics.k8s.io` API.")
+            logging.error(e)
+            return {}
+
+        return result
+
+    def get_friendly_memory_usage(self):
+        """Get nodes email-friendly memory usage."""
+        output_memory_usage = ""
+        memory_usage = self.get_memory_usage()
+        if memory_usage:
+            for node, memory in memory_usage.items():
+                output_memory_usage += f"\n  {node}: {memory.get('usage')}/{memory.get('capacity')} ({memory.get('percentage')})"
+        return output_memory_usage
+
+    def get_status(self):
+        """Get status summary for REANA nodes."""
+        return {
+            "unschedulable_nodes": self.get_unschedulable_nodes(),
+            "memory_usage": self.get_friendly_memory_usage(),
+        }
+
+
+class PodsStatus(REANAStatus):
+    """Class to retrieve statistics related to REANA cluster pods."""
+
+    def __init__(self, from_=None, until=None, user=None):
+        """Initialise PodStatus class.
+
+        :param from_: From which moment in time to collect information. Not
+            implemented yet.
+        :param until: Until which moment in time to collect information. Not
+            implemented yet.
+        :param user: A REANA-DB user model.
+        :type from_: datetime
+        :type until: datetime
+        :type user: reana_db.models.User
+        """
+        self.statuses = ["Running", "Pending", "Suceeded", "Failed", "Unknown"]
+        super().__init__(from_=from_, until=until, user=user)
+
+    def get_pods_by_status(self, status, namespace):
+        """Get pod name list by status."""
+        pods = current_k8s_corev1_api_client.list_namespaced_pod(
+            namespace, field_selector=f"status.phase={status}",
+        )
+        return [pod.metadata.name for pod in pods.items]
+
+    def get_friendly_pods_by_status(self, status, namespace):
+        """Get pod name list by status."""
+        pods = self.get_pods_by_status(status, namespace)
+
+        return "\n  ".join(["", *pods])
+
+    def get_status(self):
+        """Get status summary for REANA pods."""
+        return {
+            f"{ns}_{status.lower()}_pods": self.get_friendly_pods_by_status(status, ns)
+            for ns in self.namespaces
+            for status in self.statuses
+        }
+
+
+class JobsStatus(REANAStatus):
+    """Class to retrieve statistics related to REANA cluster jobs."""
+
+    def __init__(self, from_=None, until=None, user=None):
+        """Initialise PodStatus class.
+
+        :param from_: From which moment in time to collect information. Not
+            implemented yet.
+        :param until: Until which moment in time to collect information. Not
+            implemented yet.
+        :param user: A REANA-DB user model.
+        :type from_: datetime
+        :type until: datetime
+        :type user: reana_db.models.User
+        """
+        self.compute_backends = ["Kubernetes", "HTCondor", "Slurm"]
```

Comment:
Minor: I guess there should be some config to get these values and not to duplicate it?

Reply:
That's a very good point. Unfortunately, we save [this value in DB as a simple string](https://github.com/reanahub/reana-db/blob/01976eb2f5a9072303715d6ac6fce4ad6de3776f/reana_db/models.py#L682)... IMO it should be an enum so we can control the values assigned and in cases like this, get them all to perform queries.

I've dug a bit into our codebase and I only see [_ these values_ in reana-job-controller](https://github.com/reanahub/reana-job-controller/blob/f89bdf78b60305f271ccfe06d35ee393884eb67a/reana_job_controller/config.py#L20-L30), but we can't even use those because we can't import from this component and the names don't match exactly what we have in the database. An ideal solution IMO would be modifying this at a DB level, but it would require a schema change, alembic recipe, etc. What do you think?

I see that `r-client` also [has something similar](https://github.com/reanahub/reana-client/blob/ac31929726c4b8da7b9d8a885ed3e640b572ba18/reana_client/cli/workflow.py#L796-L800), I'll move this to `r-commons` and use it from `r-server` as a temporary workaround.

Reply:
PR created: https://github.com/reanahub/reana-client/pull/517